py3status/modules/gcal.py

```diff
@@ -0,0 +1,170 @@
+    is_event: a boolean based on available events
+
+format_event placeholders:
+    {event} scheduled event eg Pizza night
+    %?  strftime characters to be translated. See `man strftime'.
+    is_time: a boolean based on event containing a time schedule
+    is_date: a boolean based on event containing a date schedule
+
+Requires:
+    gcalcli: Google Calendar Command Line Interface
+
+Examples:
+```
+# change event colors
+gcal {
+    color_bad = '#ad7fa8'
+    color_degraded = '#fce94f'
+    color_good = '#34e2e2'
+}
+# show "No Events"
+gcal {
+    format = u'[\?if=is_event {format_event}|\?color=bad No Events]'
+}
+```
+
+@author lasers
+
+SAMPLE OUTPUT
+[
+    {'color': '#FFFF00', 'full_text': u'Fri Jul 20 '},
+    {'color': '#FF0000', 'full_text': u'Tacos'},
+    {'full_text': u', '},
+    {'color': '#FF0000', 'full_text': u'Volleyball'},
+    {'full_text': u', '},
+    {'color': '#00FF00', 'full_text': u'6:00 AM '},
+    {'color': '#FF0000', 'full_text': u'Garbage Pickup'},
+    {'full_text': u', '}
+    {'color': '#FFFF00', 'full_text': u'Sat Jul 21 '},
+    {'color': '#00FF00', 'full_text': u'6:05 PM '},
+    {'color': '#FF0000', 'full_text': u'Flight to Poland'}
+]
+"""
+
+from datetime import datetime as dt, timedelta
+import re
+
+STRING_NOT_INSTALLED = "isn't installed"
+FMT_DATE = '%a %b %d'
+FMT_GCAL = '%Y%m%d'
+FMT_SS = '%s %s'
+FMT_TIME = '%I:%M%p'
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    agenda_days = 1
+    button_refresh = 2
+    cache_timeout = 21600
+    format = u'{format_event}'
+    format_event = '[\?if=is_date&color=degraded %a %b %d ]' +\
+                   '[\?if=is_time&color=good %-I:%M %p ]' +\
+                   '[\?color=bad {event}]'
+    format_separator = ', '
+    remove = []
+
+    def post_config_hook(self):
+        self.cmd = 'gcalcli agenda --nocolor %s %s'
+        if not self.py3.check_commands(self.cmd.split(' ', 1)[0]):
+            raise Exception(STRING_NOT_INSTALLED)
+
+        remove = '|'.join(self.remove)
+        self.re = re.compile(r'(' + remove + ')')
+
+    def _get_gcal_data(self):
+        now = dt.now()
+        start = dt.strftime(now, FMT_GCAL)
+        end = dt.strftime(now + timedelta(days=self.agenda_days), FMT_GCAL)
+        return self.py3.command_output(self.cmd % (start, end))
+
+    def _organize_data(self, data):
+        new_data, is_event = [], True
+        for out in data.splitlines():
+            temporary = {}
+            if 'No Events Found' in out:
+                is_event = False
+                break
+            if out.rstrip():
+                temporary['date'] = out[:10].strip()
+                temporary['time'] = out[10:20].strip()
+                temporary['event'] = out[20:].strip()
+                temporary = {k: v for k, v in temporary.items() if v}
+                new_data.append(temporary)
+
+        return is_event, new_data
+
+    def _manipulate_data(self, data):
```

Comment:
![gcal](https://user-images.githubusercontent.com/852504/28454752-003c7f6c-6dc2-11e7-8e30-5378e0757142.png)

We parse this (w/o colors). No `%R` for time... We turn them into `datetime objects` then we make them available for `strftime`. Every line is an event so we have `is_time` and `is_date` boolean checking in `format_event` because we can have... 
* `date`, `time`, `event`
* `date`, `____`, `event`
* `____`, `time`, `event`
* `____`, `____`, `event`

I hope that clears things up. Please ask away more questions. Thx. Cheers. 🍻